### ISSUE TITLE
`match_like_matches_macro` does not trigger when one arm contains contains a block with only a bool literal

### ISSUE THREAD

**Issue by lopopolo** [rust-lang/rust-clippy#9163]
### Summary

Looking into the past discussions of `match_like_matches_macro`, it is intended for this lint to trigger when all arms contain `bool` literals. If one match arm is very long, rustfmt will wrap the `bool` literal in a block. This formatting prevents the lint from triggering

### Lint Name

match_like_matches_macro

### Reproducer

I tried this code:

```rust
fn x(c: char) -> bool {
    match c {
        '1' | '2' | '3' => {
            true
        }
        ' ' => true,
        '.' => true,
        _ => false,
    }
}

fn y(c: char) -> bool {
    match c {
        '1' | '2' | '3' => true,
        ' ' => true,
        '.' => true,
        _ => false,
    }
}

fn main() {
    dbg!(x('g'));
    dbg!(y('g'));
}
```

I expected to see this happen: both `x` and `y` trigger `clippy::match_like_matches_macro`.

Instead, this happened: only `y` triggers the lint:

```
    Checking playground v0.0.1 (/playground)
warning: match expression looks like `matches!` macro
  --> src/main.rs:13:5
   |
13 | /     match c {
14 | |         '1' | '2' | '3' => true,
15 | |         ' ' => true,
16 | |         '.' => true,
17 | |         _ => false,
18 | |     }
   | |_____^ help: try this: `matches!(c, '1' | '2' | '3' | ' ' | '.')`
   |
   = note: `#[warn(clippy::match_like_matches_macro)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#match_like_matches_macro

warning: `playground` (bin "playground") generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.50s
```

### Version

```text
Stable rustc 1.62.0
clippy 0.1.64 (2022-07-12 1c7b36d)

(playground)
```

**Comment by alex-semenyuk** [comment]
@lopopolo if you apply rustfmt on `x` it will be transfomed to `y` - [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=f6d7f4958e241e2dc486a7d2baf6a8ce)
So I think it doesn't make sense to fix it at all.

**Comment by lopopolo** [comment]
@alex-semenyuk I was being lazy when crafting the example because I was on mobile. If you add 10 or 15 more characters to the pattern, rustfmt will add the block.
